crypto/bn/bn_rsa_fips186_5.c

```diff
@@ -251,12 +259,29 @@ int ossl_bn_rsa_fips186_4_gen_prob_primes(BIGNUM *p, BIGNUM *Xpout,
     return ret;
 }
 
+static ossl_inline
+int get_multiple_of_y_congruent_to_cmod8(BIGNUM *y, const BIGNUM *r1r2x2, int c)
+{
+    int i = 0;
+
+    for (i = 0; i < 3; ++i) {
+        /* Check for Y = c mod 8 */
+        if ((int)BN_mod_word(y, 8) == c)
```

Comment:
Pretty expensive operation in the loop and it's completely avoidable:

```suggestion
        if ((int)(*bn_get_words(y) & 7) == c)
```